lib/spell-check-manager.coffee

```diff
@@ -0,0 +1,306 @@
+class SpellCheckerManager
+  checkers: []
+  locales: []
+  localePaths: []
+  useLocales: false
+  localeCheckers: null
+  knownWords: []
+  addKnownWords: false
+  knownWordsChecker: null
+
+  addPluginChecker: (checker) ->
+    console.log "spell-check: addPluginChecker:", checker
+    @addSpellChecker checker
+  addSpellChecker: (checker) ->
+    console.log "spell-check: addSpellChecker:", checker
+    @checkers.push checker
+
+  removeSpellChecker: (spellChecker) ->
+    @checkers = @checkers.filter (plugin) -> plugin isnt spellChecker
+
+  check: (args, text) ->
+    # Make sure our deferred initialization is done.
+    @init()
+
+    # We need a couple packages.
+    multirange = require 'multi-integer-range'
```

Comment:
Could you clarify on why we need the `multi-integer-range` library? 

Anyways, this is an interesting problem, as it would seem we'd need to define some rules for how the returned ranges need to be interpreted. For example: if a range is marked as incorrect, but there's an intersecting range that's marked as correct by another provider, what should we do? Is the whole range correct or incorrect? I guess we might define some importance rules in order to prevent this kind of mismatching.

I also wonder if it might be simpler to just expose a word-based API: spell checking, [as defined on Wikipedia],(https://en.wikipedia.org/wiki/Spell_checker) extracts words and then compares them against some kind of dictionary, so maybe we could avoid introducing this extra complexity by just having correct and incorrect words? Then, we should easily be able to identify which ranges are valid misspellings and whatnot.

Thoughts? :thought_balloon: